sound/soc/sof/ipc4-topology.c

```diff
@@ -447,6 +448,19 @@ static int sof_ipc4_widget_setup_pcm(struct snd_sof_widget *swidget)
 	}
 	dev_dbg(scomp->dev, "host copier '%s' node_type %u\n", swidget->widget->name, node_type);
 
+	spcm = snd_sof_find_spcm_comp(scomp, swidget->comp_id, &dir);
+	if (spcm && dir == SNDRV_PCM_STREAM_PLAYBACK) {
+		struct snd_sof_pcm_stream *sps = &spcm->stream[dir];
+
+		sof_update_ipc_object(scomp, &sps->dsp_buffer_time_ms,
+				      SOF_COPIER_DEEP_BUFFER_TOKENS,
+				      swidget->tuples,
+				      swidget->num_tuples, sizeof(u32), 1);
+		/* Set default DMA buffer size if it is not specified in topology */
+		if (!sps->dsp_buffer_time_ms)
+			sps->dsp_buffer_time_ms = SOF_IPC4_MIN_DMA_BUFFER_SIZE;
```

Comment:
ipc4-topology.h:#define SOF_IPC4_MIN_DMA_BUFFER_SIZE    2

Is this really what we want? Why would we use multiple of 2ms for the period size? I think this confuses buffer and period, no?

Reply:
@plbossart, no, the default SOF hos buffer is 2ms long (with 2x 1ms firmware periods). Initially the DMA will move 2ms on these PCMs with default buffer configuration.
The DeepBuffer PCM have by default 100ms size (with100x 1ms firmware periods), here also the DMA will move initially with the size of the DSP buffer.

That define must be renamed to `SOF_IPC4_MIN_DMA_BUFFER_TIME_MS`

Reply:
In case of capture we can allow 1ms period sizes, the firmware will push 1ms chunks afaik and we don't have DeepBuffer for capture, again, afaik.

Reply:
That doesn't answer to my question: Are we going to be limited to 2ms ALSA period sizes for playback? That is a fairly important limitation IMHO. Increases in latency are like temporary tax increases, in practice they remain permanent...

I also wonder if the firmware behavior is predicated on the use of SPIB, which prevents invalid values from being transmitted to the DSP. The problem is that SPIB is not enabled by default for PCM devices.

Reply:
> @plbossart, no, the default SOF host buffer is 2ms long (with 2x 1ms firmware periods). Initially the DMA will move 2ms on these PCMs with default buffer configuration. The DeepBuffer PCM have by default 100ms size (with100x 1ms firmware periods), here also the DMA will move initially with the size of the DSP buffer.
> 
> That define must be renamed to `SOF_IPC4_MIN_DMA_BUFFER_TIME_MS`

Is this reaslly correct? The buffer size is 2ms long alright but why would the burst be 2ms also? Doesn;'t that mean the buffer will have to twice as large ie 4ms large?

Reply:
@ranj063, if we set the constraint on the period time to be minimum of the host DMA buffer then the ALSA buffer must be at least twice as big (periods_min is 2)